packages/calypso-e2e/src/lib/utils/editor-tracks-event-manager.ts

```diff
@@ -0,0 +1,126 @@
+import { Locator, Page } from 'playwright';
+import { TracksEvent, TracksEventProperties } from '../../types';
+
+export interface TracksEventSearchOptions {
+	/**
+	 *
+	 */
+	waitForEventMs?: number;
+}
+
+// Exporting just for unit testing
+export const createEventMatchingPredicate = (
```

Comment:
As this library class evolves, I figure we'll be re-using this matching predicate in multiple potential array functions (`reduce`, `find`, `some`). I'm defining it here for re-usability, and to also be able to cleanly put it under unit test without working about a bunch of DOM and `Playwright` mocks.